subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java

```diff
@@ -70,10 +70,9 @@ public File call() {
             @Override
             public Directory call() {
                 if (usesSubprojectCheckstyleConfiguration()) {
-                    DeprecationLogger.nagUserOfDeprecated("Setting the Checkstyle configuration file under 'config/checkstyle' of a sub project", "Use the root project's 'config/checkstyle' directory instead.");
+                    DeprecationLogger.nagUserWithDeprecatedIndirectUserCodeCause("Setting the Checkstyle configuration file under 'config/checkstyle' of a sub project", "Use the root project's 'config/checkstyle' directory instead.");
```

Comment:
⭕️ These method signatures are getting really awkward. I think we should have a builder pattern instead. This doesn't block merging this PR, but would be a much appreciated follow-up.

Reply:
@breskeby please raise a card on our board to come back and do this. We will do it as part of this stream of work, so soon.

Reply:
done at https://github.com/gradle/dotcom/issues/1214